### PR TITLE
Update a couple of JavaScript packages (eslint, jsdom, moment)

### PR DESCRIPTION
**Pull request by aliceriot**
This updates eslint and the related plugin packages, turns on a few eslint rules relating to flow, and updates jsdom and moment.

Read over the changes, and assuming that you agree with the rule changes make sure that there are no regressions.

### CODE REVIEW

**Comment by aliceriot** [comment]
JSDom update was causing problems, so I'm leaving that off for now.

File: static/js/financial_aid/functions.js [hunk 1]
```diff
@@ -126,7 +126,7 @@ window.financialAidReview = (function(window, document, $) {
    */
   function initiateSearch() {
     var searchQuery = $("#search-query").val();
-    location = BASE_PATH + searchQuery;
+    location = BASE_PATH + searchQuery; // eslint-disable-line no-global-assign
```
Comment (singingwolfboy) [inline static/js/financial_aid/functions.js]: Might be clearer to do `window.location = BASE_PATH + searchQuery;`, which would avoid the need for the eslint disable.

Comment (aliceriot) [inline static/js/financial_aid/functions.js]: good idea!

**Comment by aliceriot** [comment]
ok so I hate npm but npm-shrinkwrap is sorted (T_T)

File: static/js/util/util_test.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /* global SETTINGS: false */
+/* eslint-disable no-global-assign */
```
Comment (noisecapella) [inline static/js/util/util_test.js]: You should be able to remove this since another PR today removed the global assign of `SETTINGS`

**Comment by noisecapella** [comment]
:+1: after addressing comment
